graylog2-server/src/test/java/org/graylog/plugins/pipelineprocessor/functions/FunctionsSnippetsTest.java

```diff
@@ -1665,6 +1667,27 @@ void removeFieldsByRegex() {
         assertThat(message.getField("f1")).isEqualTo("f1");
     }
 
+    @Test
+    void removeFieldsByValue() {
+        final Rule rule = parser.parseRule(ruleForTest(), true);
+        final Message message = messageFactory.createMessage("test", "test", Tools.nowUTC());
+        evaluateRule(rule, message);
+
+        assertThat(message.getField("f1")).isNull(); // match regex
+        assertThat(message.getField("f2")).isNull(); // match regex
+        assertThat(message.getField("f3")).isEqualTo("stay in message");
+        assertThat(message.getField("f4")).isNull(); // match values
+        assertThat(message.getField("f5")).isEqualTo("stay in message");
+        assertThat(message.getField("f6")).isNull(); // match values
+        assertThat(message.getField("f7")).isEqualTo("f-7");
```

Comment:
I might have missed it, but is this testing a specific condition?

Reply:
Just added it to ensure that providing a `value` doesn't behave like a `pattern`.  I added this line to the rule just to confirm it is indeed removing literal `f.7` and not treating it as a regex.
```
  remove_fields_by_value(values:["f.7"]);
```